### ISSUE TITLE
Version 22 issues

### ISSUE THREAD

**Issue by Odyseus** [BigE/desk-changer#20]
Hello, @BigE.

I tested the new version 22 of your extension and I found a couple of issues.

### Typo

There seems to be a typo on the **application.py** file in line 167. The daemon wouldn't start from the extension menu, so I launched it from the terminal and I got the following message:

```shell
Settings schema 'org.gnome.shell.extensions.desk-changer' does not contain a key named 'lockstring-profile'
```

The line reads:

```python
self.load_profile(self._settings.get_string('lockstring-profile'), True, False)
```

...when it should read...

```python
self.load_profile(self._settings.get_string('lockscreen-profile'), True, False)
```

After changing the *lockstring-profile* string to *lockscreen-profile*, the daemon started normally.

### No wallpaper rotation

With **rotation** set to an **interval**, and **random** set to false, the wallpapers aren't changed. It gets stuck on what would be the first wallpaper. Immediately after **random** is set to true, the wallpapers start to rotate. When I launch the daemon from terminal with `G_MESSAGES_DEBUG=all`, the following is part of its output.

```shell
-INFO: adding file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg to the queue
-INFO: profile default has been loaded with 19 wallpapers
(desk-changer-daemon.py:1482): -DEBUG: Profile(name=default)::preview ('file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg',)
(desk-changer-daemon.py:1482): -DEBUG: [DBUS]::preview ('file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg',)
-INFO: adding file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg to the queue
(desk-changer-daemon.py:1482): -DEBUG: Profile(name=default)::preview ('file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg',)
(desk-changer-daemon.py:1482): -DEBUG: [DBUS]::preview ('file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg',)
(desk-changer-daemon.py:1482): -DEBUG: added timer for 10 seconds
(desk-changer-daemon.py:1482): -DEBUG: [DBUS] org.freedesktop.DBus.Properties:GetAll
(desk-changer-daemon.py:1482): -DEBUG: [DBUS] org.freedesktop.DBus.Properties:GetAll
-INFO: adding file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg to the queue
(desk-changer-daemon.py:1482): -DEBUG: Profile(name=default)::preview ('file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg',)
(desk-changer-daemon.py:1482): -DEBUG: [DBUS]::preview ('file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg',)
(desk-changer-daemon.py:1482): -DEBUG: [DBUS]::changed ('file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg',)
-INFO: adding file:///usr/share/backgrounds/gnome/Bokeh_Tails.jpg to the queue

# The previous 4 lines are logged every time the daemon *switches* wallpapers.
# But it is always the exact same image.

(desk-changer-daemon.py:1482): -DEBUG: [DBUS] org.gnome.Shell.Extensions.DeskChanger.Daemon:Quit
(desk-changer-daemon.py:1482): -DEBUG: ::shutdown
(desk-changer-daemon.py:1482): -DEBUG: ::dbus_unregister
-INFO: removing DBus registration for name /org/gnome/Shell/Extensions/DeskChanger/Daemon

```

### Lock screen wallpaper rotation

I didn't find any issues with this feature because I seem not to be able to actually lock the screen. It seems that it could be due to the fact that I don't use **GDM** as my desktop manager (?), I use **LightDM** instead. I don't even have a **Lock Screen** button on my system menu.

**Comment by BigE** [comment]
Okay, I've fixed the issues you've listed here. You can try things in the develop branch until I release again to test. I'll leave this open for a bit longer for any further feedback or errors... but I'll likely make a release later tonight and close it up.

I'm considering opening another ticket for lightdm "support". I wouldn't really call it support though, just the ability to detect gdm vs lightdm and disable the "lockscreen" portions of the extension since they won't work anyway.

**Comment by Odyseus** [comment]
Tested the develop branch. All seems to be working perfectly. 👍

**Comment by BigE** [comment]
Should be fixed in 7fc9ef7 now, which is the version-23 release. Thanks for reporting the errors!
